Added UE4SS support for modding

obnyis opened the pull request:
Testing the [linux-experiment](https://github.com/Yangff/RE-UE4SS/releases/tag/linux-experiment) build of UE4SS as documented in https://github.com/UE4SS-RE/RE-UE4SS/issues/364#issuecomment-2578762122

Added new `ENABLE_UE4SS` variable to control if the binary gets downloaded, and if the server starts up with the libUE4SS.so injected.

I've currently set this up to modify/download the files `UE4SS-settings.ini`, `Mods/BPModLoaderMod/Scripts/main.lua` and `MemberVariableLayout.ini`. These should probably be added to the `configs/` directory, or at least have the download URLs updated to point to specific commit hashes.

I've done an initial test using the [Creative Menu](https://www.nexusmods.com/palworld/mods/703?tab=description) Mod, and this loads up and runs correctly. I'm not sure what long term stability is like, as I still need to apply these changes back onto my live server and test it over the week.

obnyis committed:
Added UE4SS support for modding

jammsen commented on includes/server.sh:
```diff
@@ -20,8 +20,43 @@ function start_server() {
     if [[ -n $WEBHOOK_ENABLED ]] && [[ $WEBHOOK_ENABLED == "true" ]]; then
         send_start_notification
     fi
+    if [[ -n $ENABLE_UE4SS ]] && [[ $ENABLE_UE4SS == "true" ]]; then
```
Sorry for beeing blunt/brutally honest here:

I strongly dislike this solution, this downloads a "to me/the user" unknown script and replaces the start of the server from the official script that normally starts the server.

It feels strongly like a "curl someurl.sh | sudo bash" with no idea what is happening, just pray and hope. 

This is not the way i like to code or publish code for transparency.

As you can see here  https://github.com/jammsen/docker-palworld-dedicated-server/blob/develop/Dockerfile#L5-L9 and here https://github.com/jammsen/docker-palworld-dedicated-server/blob/develop/Dockerfile#L22-L24 i try to go the Ansible way, reduce MITM and i version pin, review the code and scan the dependency with security tools, before i publish a release, because software-chain-attacks grow year over year in the millions if not billions.

I personally arent in the need of mod-support, but if the community wants this and it can be provide in a way that makes sense and is secure, im open to look into it. But this way i dont like it.

jammsen commented:
Hey @obnyis thanks for the PR, i wrote a review-comment in a lets say "very direct way" sorry if thats a bit harsh to you, i dont want to be mean, just clear about my thought about this PR. I hope you can see that. If its to harsh for you: 

Respectfully: Im very sorry.

I will ask the community in the discord if they want the feature "modding" in this image. Feel free to join in. Link is in the readme.

obnyis committed:
Refactor UE4SS support

- Move ue4ss folder to Pal/Binaries/Linux/ue4ss/
- Extract ue4ss setup from server.sh to ue4ss.sh
- Replace multiple sed / crudini calls with patch files
- Move ENABLE_UE4SS above SERVER_SETTINGS_MODE
- Add validation of downloaded files
- Download from permalinks instead of branches

jammsen commented:
As voted and discussed in Discord here: https://discord.com/channels/532141442731212810/1391461311895961712

This PR will not be continued and @obnyis + @GlitchApotamus will create another fork of my repo to add this.
